### PR TITLE
fix(docs): Update web assembly example and accepted compiler options in docs.

### LINKED ISSUE #7556: Code sample for WebAssembly doesn't compile on 1.4.0
The code sample at [Manual - Using WebAssembly](https://deno.land/manual/getting_started/webassembly) doesn't compile on Deno 1.4.0, the error message above is shown instead;

```bash
error: TS2349 [ERROR]: This expression is not callable.
  No constituent of type 'ExportValue' is callable.
console.log(wasmInstance.exports.main().toString());
```

To fix this I changed 11th line (last line) with;

```js
console.log((wasmInstance.exports.main as Function)().toString());
```

by that it shows the result.

<details><summary>Spoiler Alert</summary>
<span>42 is the result.</span>
</details>

### LINKED ISSUE #7634: docs: update TypeScript compiler options
With the update to TypeScript 4.0, there were some changes to the default compiler options (in particular we use the newer JSX options and stopped using the old deprecated ones).  The part of the documentation here: https://github.com/denoland/deno/blob/master/docs/getting_started/typescript.md#custom-typescript-compiler-options needs to be updated.

### PR DESCRIPTION
**Pull request by searchableguy**
Fixes #7634
 Fixes #7556

### CODE REVIEW

File: docs/getting_started/typescript.md [hunk 1]
```diff
@@ -153,7 +153,8 @@ Following are the currently allowed settings and their default values in Deno:
     "disableSizeLimit": false,
     "generateCpuProfile": "profile.cpuprofile",
     "jsx": "react",
-    "jsxFactory": "React.createElement",
+    "jsxFactory": "h",
```
Comment (kitsonk) [inline docs/getting_started/typescript.md:156]: This is incorrect.  Why do you think this is the default?

**Review by kitsonk — CHANGES_REQUESTED** [review]
The defaults for `jsxFactory` and `jsxFragmentFactory` are incorrect.

File: docs/getting_started/webassembly.md [hunk 2]
```diff
@@ -15,5 +15,6 @@ const wasmCode = new Uint8Array([
 ]);
 const wasmModule = new WebAssembly.Module(wasmCode);
 const wasmInstance = new WebAssembly.Instance(wasmModule);
+const main = wasmInstance.exports.main as CalleableFunction
```
Comment (caspervonb) [inline docs/getting_started/webassembly.md:18]: This example block is marked as `js`

Comment (bartlomieju) [reply docs/getting_started/webassembly.md:18]: `Callable` instead of `Calleable`?

File: docs/getting_started/typescript.md (hunk 1 above)
Comment (searchableguy) [reply docs/getting_started/typescript.md]: I updated the values from https://github.com/denoland/deno/blob/98c9798cb757045b3f46573c5baa80e2cfe672aa/cli/ast.rs#L175-L191

Is that correct now?

**Review by kitsonk — APPROVED** [review]
LGTM

**Comment by caspervonb** [comment]
Thanks @searchableguy

File: docs/getting_started/webassembly.md [hunk 3]
```diff
@@ -15,5 +15,6 @@ const wasmCode = new Uint8Array([
 ]);
 const wasmModule = new WebAssembly.Module(wasmCode);
 const wasmInstance = new WebAssembly.Instance(wasmModule);
+const main = wasmInstance.exports.main as CallableFunction
 console.log(wasmInstance.exports.main().toString());
```
Comment (pschyska) [inline docs/getting_started/webassembly.md:19]: Should this not be
````
console.log(main().toString());
````

**Pull request merged**